### ISSUE TITLE
Example does not work! Help please!

### ISSUE THREAD

**Issue by manyopensource** [byjg/jwt-wrapper#3]
ERROR: Absent authorization token

console output:
GET http://jwtexample/api.php 500 (Internal Server Error)

**Comment by manyopensource** [comment]
I have the token in my localStorage

**Comment by byjg** [comment]
The example provided requires you to run the composer. So, In order to get the example working do the follow steps:

```
cd example
composer install
php -S localhost:8080
```

**Comment by manyopensource** [comment]
The "composer install" was run. And I'm using a local server similar to xampp.

**Comment by manyopensource** [comment]
Like I said before I have the token because I've been logged in.

**Comment by byjg** [comment]
The token in this example is valid only for 60 seconds.

**Comment by manyopensource** [comment]
I checked almost immediately

**Comment by byjg** [comment]
That's strange. I double checked the example.

1. Open the client.html
2. Click on login
3. and then clicked on call API.

**Comment by manyopensource** [comment]
Yes, I did exactly like that, step by step. So if you haven't changed anything in the repo I have no idea. It may caused by my local server or something wrong with the package. Thanks for your tips!

**Comment by byjg** [comment]
I tried also running docker:

```
cd example
docker run -it --rm -p 8080:80 -v $PWD:/var/www/html byjg/php:7.2-fpm-nginx 
```

**Comment by manyopensource** [comment]
And it works?

**Comment by manyopensource** [comment]
Could it be because of the port? I didn't use port 8080.

**Comment by manyopensource** [comment]
I tested on this environment:
Php7.1
Apache

Exact data I can provide after one hour or so.

**Comment by byjg** [comment]
Well, I don't think the port is problem. I just used 8080 because I already have 80 in use.

About the PHP that should work for any version from 5.6 to 7.1

However your you ran the composer from one PHP and is running the page from another PHP (xampp) maybe the xampp does not have the dependencies that the project requires.

**Comment by manyopensource** [comment]
I will check and provide all information needed!

**Comment by byjg** [comment]
Ok. I know for sure `ext-ssl` is required. Is good to know if I am missing other requirement.

**Comment by manyopensource** [comment]
I tried to run like `php -S localhost:8080` and it works!

**Comment by manyopensource** [comment]
![image](https://user-images.githubusercontent.com/29418829/58379384-623d1a80-7fc4-11e9-9dc0-4a2c40de1b49.png)

**Comment by manyopensource** [comment]
Which port it doesn't matter but it work only if run it through console.

**Comment by manyopensource** [comment]
PHP Version 7.1.22

**Comment by manyopensource** [comment]
Apache + Nginx

**Comment by manyopensource** [comment]
Therefore I don't know yet, will it work on real domain and real server or not.

**Comment by manyopensource** [comment]
Ant it works even if I change the name of server I mean in the file include.php. I don't know is that ok or not.

**Comment by byjg** [comment]
What seems to me is that your xampp installation is missing the PHP
extension that this library requires. Try to add the ext-ssl to you xampp
installation.

**Comment by manyopensource** [comment]
Ok. I will try. And gonna tell whether it worked or not.

**Comment by byjg** [comment]
Just a follow up. Did it worked?

**Comment by manyopensource** [comment]
Nope. I've tried to find ext-ssl for Windows, but give it up. So right now I don't know what to do else.

**Comment by byjg** [comment]
OK, do you have the exact error message? 

You can get that by adding at the top of php page:
```
error_reporting(E_ALL);
ini_set('display_errors', 1);
```

or by changing the PHP.ini or even access the PHP logs.

**Comment by manyopensource** [comment]
api.php?

**Comment by byjg** [comment]
Yes. If you call directly and get an error, i would like to see that message. And I believe if you are calling the api.php from the HTML example you can get the error message by inspecting the result in your browser.

**Comment by manyopensource** [comment]
![image](https://user-images.githubusercontent.com/29418829/58560195-6e380f00-8246-11e9-9c73-7ce8775c2683.png)
![image](https://user-images.githubusercontent.com/29418829/58560261-9889cc80-8246-11e9-85a7-8da0f350a9ee.png)

**Comment by byjg** [comment]
That's error is expected when the token is missing. Are you changed the example? What is your browser version? The browser console shows any kind of error?

**Comment by manyopensource** [comment]
![image](https://user-images.githubusercontent.com/29418829/58560758-96743d80-8247-11e9-8ed4-9b4f76d0e2dc.png)

**Comment by byjg** [comment]
OK , and can you access the option Network and see what is responde body of the api.php that returned error 500?

**Comment by manyopensource** [comment]
You were right I just did `<?php var_dump($_REQUEST);die;?>` in api.php and I see an empty array

**Comment by manyopensource** [comment]
And I think this one `request.setRequestHeader('Authorization', 'Bearer ' + store.getJWT());` does'nt work

**Comment by manyopensource** [comment]
It should be here the setting of the token. Am I right?

**Comment by manyopensource** [comment]
Now I see it does'nt work even with `php -S localhost:8080`!

**Comment by byjg** [comment]
Ok, give me the output of PHP -S localhost

**Comment by manyopensource** [comment]
`PHP 7.1.22 Development Server started at Wed May 29 19:40:42 2019
Listening on http://localhost:8080
Document root is C:\OSPanel\domains\jwttest
Press Ctrl-C to quit.
[Wed May 29 19:40:46 2019] ::1:58488 [404]: /favicon.ico - No such file or directory
[Wed May 29 19:40:49 2019] ::1:58487 [200]: /
[Wed May 29 19:40:49 2019] ::1:58492 [200]: /js/store.js
[Wed May 29 19:40:49 2019] ::1:58493 [200]: /js/site.js
[Wed May 29 19:40:52 2019] ::1:58496 [200]: /login.php
[Wed May 29 19:40:53 2019] ::1:58501 [200]: /api.php
[Wed May 29 19:41:06 2019] ::1:58514 [200]: /
[Wed May 29 19:41:06 2019] ::1:58515 [200]: /js/store.js
[Wed May 29 19:41:06 2019] ::1:58517 [200]: /js/site.js
[Wed May 29 19:41:09 2019] ::1:58521 [200]: /login.php
[Wed May 29 19:41:11 2019] ::1:58524 [200]: /api.php
[Wed May 29 19:41:56 2019] ::1:58565 [200]: /login.php
[Wed May 29 19:41:57 2019] ::1:58569 [200]: /api.php`

**Comment by byjg** [comment]
There is no error 500 here :( I need the log that represents the error 500?

**Comment by manyopensource** [comment]
Ok. Give me 40 minutes.

**Comment by manyopensource** [comment]
![image](https://user-images.githubusercontent.com/29418829/58607567-f5769880-82c1-11e9-8227-09a370a1f8b3.png)

**Comment by manyopensource** [comment]
![image](https://user-images.githubusercontent.com/29418829/58607620-2b1b8180-82c2-11e9-975a-44d4a04f7e35.png)

**Comment by byjg** [comment]
OK. I want to see the `php -S localhost:8080` that matched with this error you are showing to me.

**Comment by manyopensource** [comment]
You don't undestend. Like I said if run `php -S localhost:8080` it caused any errors, only the result should be an error because any token were received by the server. But if I run normally which is `http://jwttest/` it sets token in the client side, but I get error 500 when I'm trying to "call to api"!

**Comment by manyopensource** [comment]
All information now you have written above, so you should read carefully and ask me you don't understand something. And I think you should read from github rather than from email, because I may have edited something!

**Comment by byjg** [comment]
Unfortunately I cannot make my self clear to you and seems you cannot know how to get the error message from PHP logs properly. Without the error message is causing this abnormal behavior on your installation I cannot help you. When you can provide this for me I'll be glad to help you.
